Assets/Scripts/WorldEngine/World.cs

```diff
@@ -2193,9 +2197,14 @@ public void GenerateTerrain(GenerationType type, Texture2D heightmap)
 
         ProgressCastMethod(_accumulatedProgress, "Generating Drainage Basins...");
 
+        SetCellsToEvalForDrainange();
+
         GenerateDrainageBasins();
         GenerateDrainageBasins(false); // repeat to simulate geological scale erosion
-        
+
+        _cellsToDrain.Clear();
+        NeedsDrainageRegeneration = false;
```

Comment:
What does the addition of these lines do?

Reply:
The NeedsDrainageRegeneration flag is used to indicate to the editor if the user needs to manually regenerate the world before it can be saved or simulated with. A few operations alter this flag, Brush operations (which always enable the flag), Undos/Redos (which might or might not enable the flag) and Regenerations (which always disable the flag).

The _cellsToDrain hashset is used within the GenerateDrainageBasins to figure out which cells need drainage calculations/recalculations